### ISSUE TITLE
feat: recurrence plot imaging for time series clustering

### ISSUE THREAD

**Issue by thomaschi78** [drumtorben/polars-ts#101]
## Context

Recurrence plots (RPs) convert a time series into a 2D binary/grayscale image where pixel (i,j) encodes the distance between observations x_i and x_j. This transforms the time series clustering problem into an **image similarity** problem, enabling the use of pretrained CNNs (ResNet, ViT) and standard image distance metrics (SSIM).

**Tier: 2 (moderate effort)** — new imaging module + optional vision model integration.

**References:**
- Eckmann, J.P. et al. (1987). *Recurrence Plots of Dynamical Systems*. Europhysics Letters.
- Hatami, N. et al. (2018). *Classification of Time Series by CNNs Applied to Recurrence Plots*. Signal Processing.
- Campanharo, A. & Ramos, F. (2016). *Distinguishing Time Series Using Recurrence Quantification Analysis*. JSTAT.

## Proposed API

```python
from polars_ts import to_recurrence_plot

# Convert each series to a recurrence plot image
images = to_recurrence_plot(
    df,
    threshold=0.1,       # binarization threshold (None for grayscale)
    metric="euclidean",  # point-wise distance metric
    normalize=True,      # z-normalize series first
)
# Returns: dict[str, np.ndarray] mapping unique_id → 2D array (n×n image)

# Then extract features with a pretrained model or cluster directly
from sklearn.cluster import KMeans
import numpy as np
X = np.stack([img.flatten() for img in images.values()])
labels = KMeans(n_clusters=3).fit_predict(X)
```

### Task 1: Implement `to_recurrence_plot`
- **File:** `polars_ts/imaging/recurrence.py` (new)
- For each series: compute pairwise distance matrix between all timepoints
- Optional binarization with threshold
- Optional z-normalization
- Return dict mapping series ID → 2D numpy array
- **Acceptance criteria:**
  - [ ] Output images are square (n×n for series of length n)
  - [ ] Diagonal is always 0 (self-distance)
  - [ ] Symmetric matrix
  - [ ] Threshold binarization produces {0, 1} values
  - [ ] Multi-series support
- **Scope:** Small (straightforward distance matrix computation)

### Task 2: Add recurrence quantification analysis (RQA) features
- Extract: recurrence rate, determinism, laminarity, entropy, trapping time
- These are numeric features that can feed directly into existing clustering
- **Scope:** Medium

### Task 3: Register and test
- **Scope:** Small

## Verification
```bash
uv run pytest tests/imaging/test_recurrence.py -v
```
